.github/pull_request_template.md

```diff
@@ -0,0 +1,30 @@
+<!-- Provide a general summary of your changes in the Title above -->
+
+### Description
+<!--
+Describe your changes in detail.
+For contribution process guide, look into CONTRIBUTING.md
+
+If this is a relatively large or complex change:
+ - BEFORE creating a PR, find existing or start a new discussion on Issues tab,
```

Comment:
"try finding an existing issue" or sth like that. Also, do you mean that one should respond to the existing issue or just check that the issue that can be related to when creating a PR exists?

Reply:
Re-written.

This part is for `large or complex` changes (I empasized that via `**xxx**`). The intention here was to put some thoughts and discuss the change before doing the code changes to avoid irritation of all the code changes, when discussion is happening on PR comments...